pkg/k8shandler/cluster.go

```diff
@@ -274,3 +306,72 @@ func (cState *ClusterState) amendPods(dpl *v1alpha1.Elasticsearch) error {
 	}
 	return nil
 }
+
+func updateClusterSettings(dpl *v1alpha1.Elasticsearch) error {
+	masterPods, err := listRunningMasterPods(dpl.Name, dpl.Namespace)
+	if err != nil {
+		return err
+	}
+
+	// no running elasticsearch masters were found
+	// config map already has the latest configuration
+	// all nodes spawned later will read the config map
+	if len(masterPods.Items) == 0 {
+		// in case ClusterSettingsUpdate had been requested and all master pods disapeared cancel the request
+		if err = utils.UpdateConditionWithRetry(dpl, v1alpha1.ConditionFalse, utils.UpdateUpdatingSettingsCondition); err != nil {
+			return fmt.Errorf("Unable to update Elasticsearch cluster status: %v", err)
+		}
+		return nil
+	}
+	masterPod := &masterPods.Items[0]
+
+	event, err := getClusterEvent(dpl, masterPod)
+	if err != nil {
+		return err
+	}
+
+	switch event {
+	case v1alpha1.UpdateClusterSettings:
+		if err := execClusterSettingsUpdate(dpl, masterPod); err != nil {
+			return err
+		}
+		err = utils.UpdateConditionWithRetry(dpl, v1alpha1.ConditionFalse, utils.UpdateUpdatingSettingsCondition)
+	case v1alpha1.ScaledDown:
+		err = utils.UpdateConditionWithRetry(dpl, v1alpha1.ConditionFalse, utils.UpdateScalingDownCondition)
+	case v1alpha1.ScaledUp:
+		if err := execClusterSettingsUpdate(dpl, masterPod); err != nil {
+			return err
+		}
+		err = utils.UpdateConditionWithRetry(dpl, v1alpha1.ConditionFalse, utils.UpdateUpdatingSettingsCondition)
+		err = utils.UpdateConditionWithRetry(dpl, v1alpha1.ConditionFalse, utils.UpdateScalingUpCondition)
+	case v1alpha1.NoEvent:
+		return nil
+	}
+	return err
+}
+
+func getClusterEvent(dpl *v1alpha1.Elasticsearch, pod *v1.Pod) (v1alpha1.ClusterEvent, error) {
```

Comment:
this function returns an `error` but no errors are handled and only `nil` is returned